.github/workflows/twoops-tracker-build-deploy.yml

```diff
@@ -0,0 +1,70 @@
+name: TwoopsTracker | Build and Deploy
+on:
+  push:
+    branches: [main]
+env:
+  DOKKU_REMOTE_BRANCH: "master"
+  DOKKU_REMOTE_URL: "ssh://dokku@dokku-1.dev.codeforafrica.org/twoopstracker"
+  GIT_PUSH_FLAGS: "--force"
+jobs:
+  build:
+    name: Perform CI Checks
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.10]
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Cache Pants
+      - uses: pantsbuild/actions/init-pants@c0ce05ee4ba288bb2a729a2b77294e9cb6ab66f7
```

Comment:
If no release, why not use branch names e.g. `pantsbuild/actions/init-pants@main`?

Reply:
This is mainly from the official [pant's example](https://github.com/pantsbuild/example-python/blob/main/.github/workflows/pants.yaml#L30) but, I also think using main is like using `latest` for docker images which we know could introduce breaking changes without our knowledge. Using a `SHA` is the closest we have to a `release/tag`

Reply:
This is CI @esirK . This is **exactly** where you'll want to catch any breaking change. You don't want to use `latest` on Docker images/prod dependencies because you'll only get to find out if there is a breaking change _in production_ which is too late and too costly.

PS: Checking the `pantsbuild/actions/init-pants repo`, they also use [`@main`](https://github.com/pantsbuild/actions/tree/main/init-pants#usage-example) in the example configuraiton.